wicker/core/column_files.py

```diff
@@ -178,15 +178,70 @@ def _write_column(self, column_name: str) -> None:
         self.write_buffers[column_name] = self._get_new_buffer()
 
 
-class ColumnBytesFileCache:
```

Comment:
We will move the parts of the `ColumnBytesFileCache` class that just read the column file into this new `ColumnBytesFileReader` base class (and leave the parts that cache the column file from S3 in the `ColumnBytesFileCache` class).

(Also note that in the `__init__` function that `path_factory` defaults to `S3PathFactory()` for backwards compatibility.)